rust/publish-zenodo/src/lib.rs

```diff
@@ -0,0 +1,575 @@
+    SoftwareDocumentation,
+    TaxonomicTreatment,
+    TechnicalNote,
+    Thesis,
+    WorkingPaper,
+    Other,
+}
+
+/// Publish to Zenodo
+/// 
+/// Metadata for the deposition is extracted from the metadata within the
+/// document.
+/// 
+/// Once uploaded, you will prompted to review the deposition and publish from
+/// Zenodo's web interface. If you wish to skip the review process and publish
+/// immediately, then use the `--force` flag.
+/// 
+/// By default, Zenodo's testing server is used. This means that DOIs created
+/// will not resolve. To upload to Zenodo's public-facing production server, use
+/// the `--zenodo` flag. To upload to a specific server instance, use
+/// `--server`.
+/// 
+/// Unless told otherwise, your deposit will be given the publication type "preprint".
+#[derive(Debug, Parser)]
+pub struct Cli {
+    /// Path to location of what to publish
+    #[arg(default_value = ".")]
+    #[arg(display_order(0))]
+    path: PathBuf,
+
+    /// Zenodo authentication token
+    /// 
+    /// To create one, log into Zenodo, visit your account's page, then click
+    /// "Applications", followed by "+ New Token" within the "Personal access
+    /// tokens" section. Give the token a name and enable the "deposit:actions" the scope.
+    /// 
+    /// Enable the "deposit:write" scope to enable the `--force` flag.
+    #[arg(long, env = "STENCILA_ZENODO_TOKEN")]
+    #[arg(help_heading("Zenodo Settings"))]
+    #[arg(display_order(1))]
+    token: Option<String>,
+
+    // Server selection options
+
+    /// Publish to Zenodo's testing server
+    #[arg(group = "zenodo_server")]
+    #[arg(long, default_value_t = true)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    sandbox: bool,
+
+    /// Publish to Zenodo's public-facing server
+    /// 
+    /// [default]
+    #[arg(group = "zenodo_server")]
+    #[arg(long)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    zenodo: bool,
+
+    /// Publish to a specific Zenodo server
+    #[arg(group = "zenodo_server")]
+    #[arg(long, value_parser = parse_host)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    server: Option<url::Host>,
+
+    // Resource type options
+
+    /// Upload document as a "lesson"
+    #[arg(group = "resource_type")]
+    #[arg(long)]
+    #[arg(conflicts_with_all = ["publication", "publication_type"])]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    lesson: bool,
+
+    /// Upload document as a "publication"
+    /// 
+    /// [default]
+    #[arg(group = "resource_type")]
+    #[arg(long, default_value_t = true)]
+    #[arg(default_value_if("lesson", common::clap::builder::ArgPredicate::IsPresent, "false"))]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    publication: bool,
+
+    /// Reserve a DOI for the deposition
+    #[arg(long)]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    reserve_doi: bool,
+
+    /// Publication date
```

Comment:
It is unclear what date format to use. An example here would be helpful.

Reply:
Help output now renders as 

```
      --publication-date <YYYY-MM-DD>
          Publication date
          
          Provide the date formatted as YYYY-MM-DD, e.g. 2012-03-10.
          
          When omitted, Zenodo will use today's date.
```